general/array.hpp

```diff
@@ -91,13 +95,13 @@ class Array
    template <typename CT>
    inline Array(const Array<CT> &src);
 
-   /// Deep copy from a braced init-list of convertible type
-   template <typename CT, int N>
-   explicit inline Array(const CT (&values)[N]);
-
```

Comment:
It's better to keep this constructor since the following usage is not supported by the constructor from initializer list:
```c++
   const int a[] = { 1, 2, 3 };
   Array<int> b(a);
```
Having both constructors is also consistent with what we have in class `Vector`.